### PR TITLE
Fix duplicate names of ``Pylint Messages`` in docs

### LINKED ISSUE #5740: Duplicated title in pylint's message documentation
### Current problem

Right now a title is duplicated following #5396 

### Desired solution

First named something like ``Pylint's message's types`` and second one ``Pylint's message list`` ?

### Additional context

See https://github.com/PyCQA/pylint/issues/5527#issuecomment-1024689745

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :scroll: Docs          |

## Description

Pretty stupid to have missed this. Should be merged asap to help SEO.

Closes #5740.

### CODE REVIEW

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas What are the titles of the pages in the side bar for you? For me the seconde title doesn't seem to be updated..

**Comment by Pierre-Sassoulas** [comment]
> For me the second title doesn't seem to be updated.

Same here. It's strange, although it would work as it is it look like there's no page containing "Overview" in the generated code : https://pylint--5744.org.readthedocs.build/en/5744/search.html?q=Overview

**Comment by DanielNoord** [comment]
Hmm, I'll have a look at this later.

**Review by Pierre-Sassoulas — COMMENTED** [review]
Maybe a bug in readthedoc for this MR ? We can merge anyway as "pylint messages" makes sense for ``Overview of all Pylint messages`` ?

**Comment by DanielNoord** [comment]
> Maybe a bug in readthedoc for this MR ? We can merge anyway as "pylint messages" makes sense for `Overview of all Pylint messages` ?

That would be strange.. I'd say let's wait to find the cause of the issue before merging.

**Comment by DanielNoord** [comment]
Fixed. Stupid mistake... The page was actually auto-generated, so changing the title in the file does nothing 😓

**Comment by Pierre-Sassoulas** [comment]
> The page was actually auto-generated,

Should we had warning as comment at the beginning of the file ?

**Comment by DanielNoord** [comment]
I don't think we should check for it being up to date, that would mean every change to a message would add changes to the file. Which would inflate commit sizes unnecessarily. I'll add a comment though!

**Comment by Pierre-Sassoulas** [comment]
> I don't think we should check for it being up to date,

Could you add a release stage in tbump to make sure it's up to date at release time then ?

**Comment by DanielNoord** [comment]
> > I don't think we should check for it being up to date,
> 
> Could you add a release stage in tbump to make sure it's up to date at release time then ?

Do we want it to be? (I have no experience with `tbump` so no idea how to implement this).

**Comment by Pierre-Sassoulas** [comment]
> Do we want it to be? 

I think it's better if it's not too far from the current state of the code. Another positive things is that the diff is a nice summary of what message appeared between two releases.

>  I have no experience with tbumb

I made the change myself let me know what you think.

File: doc/exts/pylint_messages.py [hunk 1]
```diff
@@ -122,7 +122,10 @@ def _write_messages_list_page(
         stream.write(
             f""".. _messages-list:
 
-{get_rst_title("Pylint Messages", "=")}
+.. NOTE: This file is auto-generated. Make any changes to the associated
+.. docs extension in '{__name__}.py'.
```
Comment (DanielNoord) [inline doc/exts/pylint_messages.py:126]: Smart!

**Review by DanielNoord — COMMENTED** [review]
Makes sense! Thanks for your help!

**Comment by DanielNoord** [comment]
Build is failing though :( Do you see what's wrong?

**Comment by Pierre-Sassoulas** [comment]
> Build is failing though 

Hurgh, sorry it was working locally. It sounds like the problem is ``doc/messages/messages_introduction.rst:15: WARNING: Failed to create a cross reference. A title or caption not found: messages-list`` I dont know what I changed the broke that. I'm going to check that tomorrow.

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Have you had the time to look at this? We should probably try and merge this soon. I can also take a look?

**Comment by Pierre-Sassoulas** [comment]
I have no idea right now, and I have a busy week.

What I tried :
- tox -e docs (work perfectly locally)
- trying to launch manually (also works)
- Changing the comment syntax near the reference (did not work)

What I thought about doing next:
- Bisection of the change in github actions to see what is causing the problem exactly (but it's not elegant)

Ideas welcomed :)

**Comment by Pierre-Sassoulas** [comment]
Damn, it was this 'simple' in the end. Thank you for the debug work @DanielNoord, I'm under the water right now and appreciate that.

**Comment by DanielNoord** [comment]
Same, this was an easy fix I could do in between jobs 😄

**Pull request merged**